alf/algorithms/ddpg_algorithm.py

```
@@ -237,10 +237,12 @@ def _sample(a, ou):
                                           noisy_action, self._action_spec)
         state = empty_state._replace(
             actor=DdpgActorState(actor=state, critics=()))
+        # action_distribution is not supported for continuous actions for now.
+        # Returns empty action_distribution to fail early.
         return AlgStep(
             output=noisy_action,
             state=state,
-            info=DdpgInfo(action=noisy_action, action_distribution=action))
+            info=DdpgInfo(action=noisy_action, action_distribution=()))
```

Why do you need this change? By default we could think of a deterministic action distribution as an action tensor.

---

This is to fail early and clearly.  action is a tensor, not distribution.  Putting action directly there could cause confusion when debugging.  See comment 3 lines above.

---

It is intended to return `action_distribution` here so that some other algorithm can use it (e.g. TracAlgorithm). Do you find it causing problem?

---

Yes, when used with e.g. Retrace, a distribution is needed, but action is not a distribution.